python/cudf/cudf/_lib/sort.pyx

```diff
@@ -141,7 +145,12 @@ cdef pair[vector[order], vector[null_order]] ordering(
 
 
 @acquire_spill_lock()
-def order_by(list columns_from_table, object ascending, str na_position):
+def order_by(
```

Comment:
I would weakly prefer two separate functions, one stable and one unstable. The function name explicitly carries this information in C++ and mirroring that in Cython would be ideal in my view. Otherwise you get confusing questions about what `True` or `False` means in the signature, and there's an undesirable "default" choice that will be inevitably misused.

Another option that gives it more verbosity would be to have a _required_ parameter with a string like `stable="stable"` / `stable="unstable"` but that's gross...

(Same for `[stable_]sort_by_key` below.)

Reply:
I can use required keyword args:

```python
def order_by(..., *, stable):
    ...
```
Then `order_by(..., stable=True)` or `order_by(...., stable=False)` are the only calls that are offered and are reasonably self-documenting at the call site. WDYT?

Reply:
I’m fine with a required kwarg with no default.

Reply:
Used kwargs, and made the callers default to `stable=True` if we are in pandas-compatible mode, and `stable=False` otherwise.